### ISSUE TITLE
Requests seems stuck when enabling restrictions (Quota or Rate limiting) on a plan

### ISSUE THREAD

**Issue by micedre** [gravitee-io/issues#4175]
### Expected Behavior

When enabling quota or rate limiting on a plan and then deploying the new plan, I expect to have a response from the endpoint or a response saying the quota is expired.

### Current Behavior

I've got no response, the request seems stuck in the gateway and I receive a 504 from my Load Balancer (which fronts the gateway) after 1 minutes

### Steps to Reproduce (for bugs)

1. Add an api to https://api.gravitee.io/echo and a plan with api-key security (like here) https://docs.gravitee.io/apim/3.x/apim_quickstart_publish.html
2. Verify it works by subscribing and sending a request : 
```
> curl "https://gateway-url/echo-test" -H "X-Gravitee-Api-Key: 0ad7a64e-[...]03c6b"
{
  "headers" : {
    "Host" : "api.gravitee.io",
    "User-Agent" : "curl/7.67.0",
    "Accept" : "*/*",
    "Accept-Encoding" : "deflate, gzip",
    "X-Forwarded-For" : "...",
    "X-Forwarded-Host" : "api.gravitee.io",
    "X-Forwarded-Port" : "443",
    "X-Forwarded-Proto" : "https",
    "X-Forwarded-Server" : "2190b6ef1a19",
    "X-Gravitee-Api-Key" : "0ad7a64e-[...]03c6b",
    "X-Gravitee-Request-Id" : "274f3573-bc3f-4930-8f35-73bc3f7930a1",
    "X-Gravitee-Transaction-Id" : "274f3573-bc3f-4930-8f35-73bc3f7930a1",
    "X-Real-Ip" : "...."
  },
  "query_params" : { }
}
```
3. Enable quota limiting on the plan
4. Retry sending the request, it times out.

I tried refreshing the subscription, or testing with a new keyless plan, same behavior. 

The logs when processing the request : 
```
06:30:36.398 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.reactor.impl.DefaultReactor - Receiving a request a422e370-8d92-4e02-a2e3-708d925e0266 for path /echo-test
06:30:36.411 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.s.a.ApiKeyAuthenticationHandler - Looking for an API Key from request header: X-Gravitee-Api-Key
06:30:36.413 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.s.core.SecurityPolicyResolver - Authentication handler [api_key] has been selected to secure incoming request a422e370-8d92-4e02-a2e3-708d925e0266
06:30:36.416 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.policy.impl.PolicyFactoryImpl - Create a new policy instance for io.gravitee.policy.apikey.ApiKeyPolicy
06:30:36.417 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.policy.impl.PolicyFactoryImpl - Looking for a constructor to inject policy configuration
06:30:36.418 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.p.impl.DefaultPolicyManager - Policy api-key has been added to the policy chain
06:30:36.429 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.policy.impl.PolicyFactoryImpl - Create a new policy instance for io.gravitee.policy.quota.QuotaPolicy
06:30:36.429 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.policy.impl.PolicyFactoryImpl - Looking for a constructor to inject policy configuration
06:30:36.430 [vert.x-eventloop-thread-0] [] DEBUG i.g.g.p.impl.DefaultPolicyManager - Policy quota has been added to the policy chain
06:30:36.851 [vert.x-eventloop-thread-0] [] WARN  i.g.g.h.a.p.plan.PlanPolicyResolver - No plan has been selected to process request a422e370-8d92-4e02-a2e3-708d925e0266. Returning an unauthorized HTTP status (401)
```

### Your Environment

I'm using gravitee with docker images and the `latest` tag (so `3.1.2` at the moment). My gateway is fronted by a load balancer (HaProxy). I'm using mongodb and elastic search with the following env variables : 

```
 "GRAVITEE_MANAGEMENT_MONGODB_URI": "mongodb://mongo_host:27017/gravitee",
    "GRAVITEE_REPORTERS_ELASTICSEARCH_ENDPOINTS_0": "http://elastic_host:9200",
    "GRAVITEE_RATELIMIT_MONGODB_URI": "mongodb://mongo_host:27017/gravitee",
 
```

The attached file [gravitee-stdout.log](https://github.com/gravitee-io/issues/files/5049485/gravitee-stdout.log) is a complete log from gateway start with the following events : 
07:16:21.726 : a request to the api with a plan without restrictions
07:16:45.017 : update to the plan to add quota limiting 
07:16:45.250 : new request with same api-key (with no response)

**Comment by micedre** [comment]
It works when I add the policy in the design page (if disabled in the plan) : 
![image](https://user-images.githubusercontent.com/243027/89777398-e74e1a00-db0b-11ea-9f7e-cff5bfeb7c34.png)

**Comment by leleueri** [comment]
Same issue as gravitee-io/issues#4169 
Will be fixed when 3.0.7 will be merged with branch 3.1.x

**Comment by tcompiegne** [comment]
Fixes by https://github.com/gravitee-io/issues/issues/4194
